workstreams/Adaptive_Profile_Workstream.md

```diff
@@ -0,0 +1,160 @@
+raised as early as possible with the steering committee.
+
+## Dependencies & Related Projects
+
+A related workstream is to specify entry point functions including runtime
+initialization and finalization in the QIR specification. Any rules for entry
+points defined as part of this profile should be compliant with that
+specification. The specification of an Adaptive Profile should also naturally
+extend the Base Profile specification that has already been defined.
+
+## Deliverable(s) & Expected Outcome
+
+The expected outcome is a document that clearly defines an Adaptive Profile
+meeting the requirements outlined [above](#requirements). The intent is for the
+Adaptive Profile to permit expressing quantum programs that contain classical
+computations with certain data types and non-trivial control flow structures. It
+will limit expressiveness to guarantee that the program always terminates (no
+unbounded loops or recursions). This specification will replace what is
+currently drafted as [Profile
+B](https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/7_Profiles.md#profile-a-basic-quantum-functionality)
+on the [specification repository](https://github.com/qir-alliance/qir-spec).
+
+Like the Base Profile, the Adaptive Profile should not rely on any particular
+quantum instruction(s) being available. To target a quantum program to a
+specific backend requires selecting a suitable profile and a compatible quantum
+instruction set. The profile and the quantum instruction set (QIS) selection
+together determine how the program IR should be represented.
+
+A similar specification for the Base Profile can be found
+[here](https://github.com/qir-alliance/qir-spec/blob/main/specification/under_development/profiles/Base_Profile.md).
+The specification of the Adaptive Profile should follow a similar outline for
+the sake of consistency and comprehensiveness of the specification.
+
+## Future Work (out of scope for the Adaptive Profile)
+
+The following features will not be supported in programs that are compliant with
+the Base Profile:
+
+- Composite data types: <br/>
+  The use of composite data types such as [structure
+  types](https://llvm.org/doxygen/group__LLVMCCoreTypeStruct.html) including
+  [tuples](https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/1_Data_Types.md#tuples-and-user-defined-types),
+  and [sequential
+  types](https://llvm.org/doxygen/group__LLVMCCoreTypeSequential.html) including
+  [arrays](https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/1_Data_Types.md#arrays)
+  is not supported within an Adaptive Profile. This also precludes the use of
+  such data structures for [callable
+  values](https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/2_Callables.md);
+  i.e., the usage of subroutines as first class values is not supported within
+  the Base Profile, and the use of [function
```

Comment:
Likewise, did you mean `Adaptive Profile` here?

Reply:
Yes, thank you.